tools/Performance/DynamoPerformanceTests/ResultsComparer.cs

```diff
@@ -0,0 +1,167 @@
+
+        /// <summary>
+        /// Creates a results comparer to compare results at basePath and diffPath
+        /// </summary>
+        /// <param name="basePath"></param>
+        /// <param name="diffPath"></param>
+        public ResultsComparer(string basePath, string diffPath)
+        {
+            BaseResults = ImportResultsCSV(basePath);
+            DiffResults = ImportResultsCSV(diffPath);
+
+            Compare();
+        }
+
+        /// <summary>
+        /// Compare the two results files and build comparison dictionary
+        /// </summary>
+        private void Compare()
+        {
+            Comparison = new Dictionary<string, BenchmarkResult>();
+
+            foreach(var result in DiffResults.Values)
+            {
+                var baseResult = GetBaseResult(result);
+                Comparison[result.Method + result.Graph] = new BenchmarkResult()
+                {
+                    Method = result.Method,
+                    Graph = result.Graph,
+                    Mean = result.Mean / baseResult.Mean,
+                    Error = result.Error / baseResult.Error,
+                    StdDev = result.StdDev / baseResult.StdDev,
+                };
+            }
+        }
+
+        public void WriteResultsCSV(string filePath)
+        {
+            var csv = new StringBuilder();
+            csv.AppendLine("Method,Graph,Version,Mean,Error,StdDev");
+            foreach(var result in DiffResults.Values)
+            {
+                var baseResult = GetBaseResult(result);
+                var baseLine = new string[] { result.Method, result.Graph, "Base", baseResult.Mean.ToString(), baseResult.Error.ToString(), baseResult.StdDev.ToString() };
+                csv.AppendLine(string.Join(",", baseLine));
+
+                var diffLine = new string[] { "", "", "Diff", result.Mean.ToString(), result.Error.ToString(), result.StdDev.ToString() };
+                csv.AppendLine(string.Join(",", diffLine));
+
+                var compResult = Comparison[result.Method + result.Graph];
+                var compLine = new string[] { "", "", "", compResult.Mean.ToString(), compResult.Error.ToString(), compResult.StdDev.ToString() };
+                csv.AppendLine(string.Join(",", compLine));
+            }
+
+            filePath = GetFullPath(filePath);
+            filePath += "/comparison.csv";
+            File.WriteAllText(filePath, csv.ToString());
+        }
+
+        private BenchmarkResult GetBaseResult(BenchmarkResult result)
+        {
+            var key = result.Method + result.Graph;
+            if (BaseResults.ContainsKey(key))
+            {
+                return BaseResults[key];
+            }
+            return null;
+        }
+
+        private Dictionary<string, BenchmarkResult> ImportResultsCSV(string csvPath)
+        {
+            csvPath = GetFullPath(csvPath);
+            TextFieldParser parser = new TextFieldParser(csvPath);
```

Comment:
@mjkkirschner @scottmitchell This is depending on VB, is there alternative?

Reply:
@mjkkirschner @QilongTang I'll find an alternative. The problem I was trying to solve here was parsing a csv with values that contain commas. Probably a way to regex it.